core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -500,4 +501,21 @@ public static Snapshot latestSnapshot(TableMetadata metadata, String branch) {
 
     return metadata.snapshot(ref.snapshotId());
   }
+
+  /**
+   * Checks if a particular branch of the table may contain deletes based on snapshot summary.
+   *
+   * @param table a {@link Table}
+   * @param branch branch name of the table (nullable)
+   * @return true if the branch may contain deletes
+   */
+  public static boolean mayHaveDeletes(Table table, String branch) {
+    Snapshot snapshot = latestSnapshot(table, branch);
+    if (snapshot == null) {
+      return false;
+    }
+
+    String value = snapshot.summary().get(SnapshotSummary.TOTAL_DELETE_FILES_PROP);
```

Comment:
Should we note here this is not a spec authorized field. A client is not required to put in TOTAL_DELET_FILES_PROP with the actual total number of delete files (although it would be weird if they did)

Reply:
That's a good idea, I'll add a note.

Reply:
Added.